### ISSUE TITLE
Long cannot be cast to Integer

### ISSUE THREAD

**Issue by stakach** [couchbaselabs/elasticsearch-transport-couchbase#153]
We are using ElasticSearch 2.4.3 and Transport plugin 2.2.4.3

Seems to be related to this line of code: https://github.com/couchbaselabs/elasticsearch-transport-couchbase/blob/master/src/main/java/org/elasticsearch/transport/couchbase/capi/ElasticSearchCAPIBehavior.java#L372

```
[2017-03-17 14:41:21,621][WARN ][org.eclipse.jetty.servlet.ServletHandler] /aca%2f708/_bulk_docs
java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
        at org.elasticsearch.transport.couchbase.capi.ElasticSearchCAPIBehavior.bulkDocs(ElasticSearchCAPIBehavior.java:373)
        at com.couchbase.capi.servlet.CAPIServlet.handleBulkDocs(CAPIServlet.java:529)
        at com.couchbase.capi.servlet.CAPIServlet.service(CAPIServlet.java:89)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:542)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:499)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
        at java.lang.Thread.run(Thread.java:745)
```

Any way to avoid this? I noticed that this was an issue a few years ago. #52

**Comment by stakach** [comment]
Given this is the extent of the code and expiration is cast to a long anyway - is there any harm in changing the line to `Number expiration = (Number)meta.get("expiration");`

```java
            long ttl = 0;
            Integer expiration = (Integer)meta.get("expiration");
            if(expiration != null && expiration > 0) {
                ttl = (expiration.longValue() * 1000) - System.currentTimeMillis();
                if(logger.isDebugEnabled())
                    logger.debug(String.format("Document %s has expiration set, which is not supported in ES 5.0+", id));
            }
```

**Comment by stakach** [comment]
This change solved all our issues - is this something that could be included into main?
Happy to make a pull request.

```java
            long ttl = 0;
            Object temp = meta.get("expiration");
            if (temp instanceof Long) {
                Long expiration = (Long)meta.get("expiration");
                if(expiration != null && expiration > 0) {
                    ttl = (expiration.longValue() * 1000) - System.currentTimeMillis();
                }
            } else {
                Integer expiration = (Integer)meta.get("expiration");
                if(expiration != null && expiration > 0) {
                    ttl = (expiration.longValue() * 1000) - System.currentTimeMillis();
                }
            }
```

**Comment by dnault** [comment]
@stakach Thank you for the bug report and the fix, and sorry for the belated response. This will be fixed in the upcoming 3.0.0 release (for ES versions 2.2.0 through 5.6.4).
